internal/resource-request-operator/utils.go

```diff
@@ -0,0 +1,73 @@
+package resourceRequestOperator
+
+import (
+	"context"
+	discoveryv1alpha1 "github.com/liqotech/liqo/apis/discovery/v1alpha1"
+	sharingv1alpha1 "github.com/liqotech/liqo/apis/sharing/v1alpha1"
+	"github.com/liqotech/liqo/pkg/discovery"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"time"
+)
+
+//this function generate an empty offer
+func (r *ResourceRequestReconciler) generateResourceOffer(request *discoveryv1alpha1.ResourceRequest) error {
+	err := r.computeResources()
+	if err != nil {
+		return err
+	}
+
+	offer := &sharingv1alpha1.ResourceOffer{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: request.GetNamespace(),
+			Name:      offerPrefix + r.ClusterId,
+		},
+	}
+
+	op, err := controllerutil.CreateOrUpdate(context.Background(), r.Client, offer, func() error {
+		if offer.CreationTimestamp.IsZero() {
+			//immutable labels
+			offer.Labels = map[string]string{
+				discovery.ClusterIdLabel: request.Spec.ClusterIdentity.ClusterID,
+			}
+
+			//set owner reference
+			err = controllerutil.SetOwnerReference(request, offer, r.Scheme)
```

Comment:
This could be moved to the last line, as it is probably better leave error management out of the mutation function 

```
return controllerutil.SetOwnerReference(request, offer, r.Scheme)
```